OpenRA.Mods.Cnc/Traits/MadTank.cs

```diff
@@ -56,8 +56,19 @@ class MadTankInfo : ITraitInfo, IRulesetLoaded, Requires<ExplodesInfo>, Requires
 		public object Create(ActorInitializer init) { return new MadTank(init.Self, this); }
 		public void RulesetLoaded(Ruleset rules, ActorInfo ai)
 		{
-			ThumpDamageWeaponInfo = rules.Weapons[ThumpDamageWeapon.ToLowerInvariant()];
-			DetonationWeaponInfo = rules.Weapons[DetonationWeapon.ToLowerInvariant()];
+			WeaponInfo thumpDamageWeapon;
+			WeaponInfo detonationWeapon;
+			var thumpDmgWeapToLower = ThumpDamageWeapon.ToLowerInvariant();
```

Comment:
Style nit: avoid removing arbitrary letters from variable names.  Saving 5 characters on a throwaway local variable isn't worth the hit to readability.  `thumpDamageWeaponToLower` and `detonationWeaponToLower` are fine.  If you really want to use shorter names then `thumpName` and `detonationName` would also be fine considering the context.

Reply:
I suggest using `(ThumpDamageWeapon ?? string.Empty).ToLowerInvariant()` here and in the other cases where the weapons aren't optional to avoid a `NullReferenceException` if the mod yaml overrides it to be empty (which translates to a null value).